cosmoz-grouped-list.js

```diff
@@ -205,48 +196,48 @@
 		 * @returns {void|array} Prepared data.
 		 */
 		_prepareData(data = null) {
-			if (data === null || data.length === 0 || data[0] === undefined) {
```

Comment:
Considering the data.reduce on row 202, we should not remove the null check/array check here?

Reply:
Of course, you are right.
I think that we should reify the requirement that data is an array and actually throw an Error if the expectation is unmet. This would ensure that the code is used correctly by the developers.

`this._assertIsArray(data)`

Also, empty array is a valid case and the code should run it's course (for instance, setting empty data would also clear the selection).

Reply:
@JockeCK what do you think?

Reply:
Great find @JockeCK !
Well due to the nature of data bindings and observers, we normally want methods to be able to run on "partial" states and gracefully exit rather than throw Errors.
So let's aim to exit if the method cannot complete and we'll look into where we can throw errors.

Reply:
I agree with Nomego